pkg/server/container_stats_list_windows.go

```diff
@@ -109,13 +109,31 @@ func (c *criService) getSandboxMetrics(
 		}
 		if s, ok := v.(*runhcsstats.Statistics); ok {
 			timestamp := stats.Timestamp.UnixNano()
-			cs.Cpu = &runtime.CpuUsage{
-				Timestamp:            timestamp,
-				UsageCoreNanoSeconds: &runtime.UInt64Value{Value: s.VM.Processor.TotalRuntimeNS},
-			}
-			cs.Memory = &runtime.MemoryUsage{
-				Timestamp:       timestamp,
-				WorkingSetBytes: &runtime.UInt64Value{Value: s.VM.Memory.WorkingSetBytes},
+			// If the pod is process isolated VM will be nil. We have special treatment
+			// for hypervisor isolated pod stats as we return the UVMs working set and
+			// other metrics instead of the actual sandbox containers. This also avoids
+			// a nil pointer dereference on trying to use the VM stats for process isolated.
+			if s.VM != nil {
+				cs.Cpu = &runtime.CpuUsage{
+					Timestamp:            timestamp,
+					UsageCoreNanoSeconds: &runtime.UInt64Value{Value: s.VM.Processor.TotalRuntimeNS},
+				}
+				cs.Memory = &runtime.MemoryUsage{
+					Timestamp:       timestamp,
+					WorkingSetBytes: &runtime.UInt64Value{Value: s.VM.Memory.WorkingSetBytes},
+				}
+			} else {
+				// Only have to check windows as LCOW will be hypervisor isolated
+				if windowsStats := s.GetWindows(); windowsStats != nil {
```

Comment:
I think it would be cleaner to pull out the container stats code (both here and above on lines 65..83) and put it in a separate function (e.g. `getSingleContainerMetrics`).